data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -383,4 +389,31 @@ public void deleteEnvironmentRevision(@NonNull final EnvironmentRevision environ
           e);
     }
   }
+
+  @Override
+  public List<Cluster> listClusters(String accountId, String clusterNamePrefix) {
+    if (clusterNamePrefix != null) {
+      Validate.notNull(accountId, "accountId must be specified if clusterNamePrefix is specified");
```

Comment:
Not sure if we want to enforce an accountId if clusterNamePrefix is provided.  The `listCluster` is mostly used by reconciler to schedule work per cluster.  With only clusterNamePrefix, it's a way to partition the clusters and potential run multiple reconciler with different assigned name prefix.

Reply:
This API is based on the assumption that the data model will have accountID as hash key and cluster name as range key. In that schema, DDB will not allow queries for only the range key without scanning.

I suspect that we'll have to partition along some completely synthetic axis (a consistent hash perhaps?) if we do partition here, so maybe even (accountId, clusterName) isn't the right key schema.

At least for tests, allowing this query pattern is pretty handy.

Reply:
y, make sense.